brave/src/main/java/com/linecorp/armeria/server/brave/ArmeriaHttpServerAdapter.java

```diff
@@ -0,0 +1,148 @@
+        return INSTANCE;
+    }
+
+    private ArmeriaHttpServerAdapter() {
+    }
+
+    @Override
+    public String method(RequestLog requestLog) {
+        return requestLog.method().name();
+    }
+
+    /**
+     * Original implementation is calling {@link HttpServerAdapter#url(Object)} which needs
+     * {@link RequestLog#scheme()} is not available at {@link RequestLogAvailability#REQUEST_START}.
+     * We need to use {@link RequestLog#path()} directly.
+     *
+     * @see brave.http.HttpAdapter#path(Object)
+     */
+    @Override
+    public String path(RequestLog requestLog) {
+        return requestLog.path();
+    }
+
+    @Override
+    @Nullable
+    public String url(RequestLog requestLog) {
+        return SpanTags.generateUrl(requestLog);
+    }
+
+    @Override
+    @Nullable
+    public String requestHeader(RequestLog requestLog, String name) {
+        if (!requestLog.isAvailable(RequestLogAvailability.REQUEST_HEADERS)) {
+            return null;
+        }
+        return requestLog.requestHeaders().get(name);
+    }
+
+    @Override
+    @Nullable
+    public Integer statusCode(RequestLog requestLog) {
+        final int result = statusCodeAsInt(requestLog);
+        return result != 0 ? result : null;
+    }
+
+    @Override
+    public int statusCodeAsInt(RequestLog requestLog) {
+        if (!requestLog.isAvailable(RequestLogAvailability.RESPONSE_HEADERS)) {
+            return 0;
+        }
+        return requestLog.status().code();
+    }
+
+    /**
+     * Returns the authority of the {@link Request}.
+     */
+    @Nullable
+    public String authority(RequestLog requestLog) {
+        if (!requestLog.isAvailable(RequestLogAvailability.REQUEST_HEADERS)) {
+            return null;
+        }
+        return requestLog.authority();
+    }
+
+    /**
+     * Returns the {@link SessionProtocol#uriText()} of the {@link Request}.
+     */
+    @Nullable
+    public String protocol(RequestLog requestLog) {
+        if (!requestLog.isAvailable(RequestLogAvailability.SCHEME)) {
+            return null;
+        }
+        return requestLog.scheme().sessionProtocol().uriText();
+    }
+
+    /**
+     * Returns the {@link SerializationFormat#uriText()} if it's not {@link SerializationFormat#NONE}.
+     */
+    @Nullable
+    public String serializationFormat(RequestLog requestLog) {
+        if (!requestLog.isAvailable(RequestLogAvailability.SCHEME)) {
+            return null;
+        }
+        final SerializationFormat serFmt = requestLog.scheme().serializationFormat();
+        return serFmt == SerializationFormat.NONE ? null : serFmt.uriText();
+    }
+
+    /**
+     * Returns the method name if {@link RequestLog#requestContent()} is {@link RpcRequest}.
+     */
+    @Nullable
+    public String rpcMethod(RequestLog requestLog) {
+        if (!requestLog.isAvailable(RequestLogAvailability.REQUEST_CONTENT)) {
+            return null;
+        }
+        final Object requestContent = requestLog.requestContent();
```

Comment:
similar comments here in the server call. note that it is probably higher priority to support http.route vs support rpc.method.

We can add rpc.method span name decorator easily on top.  like this https://github.com/openzipkin/brave/blob/master/instrumentation/spring-webmvc/src/main/java/brave/spring/webmvc/SpanCustomizingAsyncHandlerInterceptor.java

Reply:
you mean in the `BraveService`? Also what do you mean `http.route vs rpc.method`. About http service it's good to use http.route. and rpc.method should be used if it's rpc?

Reply:
adapter/parser is using order of `route -> methodFromResponse -> method(http)` 
And I want to overwrite it and before span#finish.

Reply:
I think best to not focus too much on rpc.method right now. let's make this work with route first. the order problem you mention deserves its own change. Also remember a user can add a tag with the rpc method regardless. It isn't required to be made a part of the span name.

Reply:
We can use `RequestLog.context().route().meterTag()` as the value of `http.route`. Not sure if we can fill this for the client side though.

Reply:
@trustin you mean we will add `route` in `RequestLog#context`?

@adriancole I see, so keep this PR as is?

Reply:
Didn't see the method in the context..

Reply:
It would be awesome if this can land with http route as the experience will be much better. ex in zipkin all our server spans are get or post. If that's too hard we can do later. regardless, let's do rpc method later

Reply:
ah...service request context..let me check if i can add in this PR directly.